File: eds-starter-archetype/src/main/resources/archetype-resources/src/main/java/entity/Role.java
#set( $symbol_pound = '#' )
#set( $symbol_dollar = '$' )
#set( $symbol_escape = '\' )
package ${package}.entity;

import javax.persistence.Entity;
import javax.validation.constraints.NotNull;
import javax.validation.constraints.Size;

import ch.ralscha.extdirectspring.generator.Model;

@Entity
@Model(value = "E4ds.model.Role", readMethod = "userService.readRoles")
public class Role extends AbstractPersistable {

	@NotNull
	@Size(max = 50)
	private String name;

	public String getName() {
		return name;
	}

	public void setName(String name) {
		this.name = name;
	}

}

File: eds-starter-archetype/src/main/resources/archetype-resources/src/main/webapp/app/model/Role.js
Ext.define("E4ds.model.Role",
{
  extend : "Ext.data.Model",
  fields : [ {
    name : "name",
    type : "string"
  }, {
    name : "id",
    type : "int"
  } ],
  proxy : {
    type : "direct",
    directFn : "userService.readRoles"
  }
});